### DISCUSSION TITLE
Can / should I disable Caddy?

### DISCUSSION THREAD

**Discussion by donferi** [php/frankenphp#1008] [Q&A] [answered]
This is probably a stupid question but if I have TLS termination and a proxy already, is there a need to have Caddy?

Couldn't my proxy talk to franken directly?

I'm sure I'm missing something here, can someone please ELI5?

**Comment by withinboredom** [comment]
You can disable TLS on the frankenphp side, and do your TLS termination elsewhere.

Frankenphp is shipped built into caddy, as a caddy module, but with a bit of work, you can strip out all "webserver" functionality by reducing the caddy config down to "php only". If you want some help with that, let us know. 

In any case, you need some kind of server (called a SAPI) to serve php code, such as Apache, php-fpm. Frankenphp replaces those older SAPIs with something more modern, but is more comparable to Apache than fpm, but just as (if not more) performant as a well tuned fpm.

Hopefully that answers your question?

**Reply by donferi** [reply]
Thank you so much for your explanation, if I understood this correctly its closer to something like nginx unit running php where there is a single container / process so no need for supervisor or s6?

I couldn't find in the docs how to disable `TLS`, could you please point me in the right direction? Thank you again!

**Reply by withinboredom** [reply, marked as answer]
To disable TLS, the most ideomatic thing to do is to set `SERVER_NAME=:8080`. Note the colon then the port number. When set to a single port (like this), Caddy serves http-only.
